### DISCUSSION TITLE
how to use code splitting the right way

### DISCUSSION THREAD

**Discussion by chj-damon** [callstack/repack#84] [Q&A] [answered]
in web projects, we can generate chunks based on route, in react-native projects, we use `react-navigation` for example, How can I generate different chunks based on that?

I read the `webpack.config.js`， the entry is just: 
```js
  /**   * `getInitializationEntries` will return necessary entries with setup and initialization code.   * If you don't want to use Hot Module Replacement, set `hmr` option to `false`. By default,   * HMR will be enabled in development mode.   */  
entry: [    
    ...ReactNative.getInitializationEntries(reactNativePath, {      
        hmr: devServer.hmr,    
    }),    
    entry,  
],
```

**Comment by zamotany** [comment, marked as answer]
The easiest way is to do it in code. I'm not sure what do you mean _generate chunks based on route_ exactly but if you're using `react-navigation` you should be able to do something like this:

```js
// MyScreen.js
function MyScreen() {
  // ...
}

// somewhere
const MyScreen = React.lazy(() => import('./MyScreen.js'));

<Stack.Navigator>
  <Stack.Screen name="MY_SCREEN" component={MyScreen} />
</Stack.Navigator>
```

**Reply by chj-damon** [reply]
Thanks, that's exactly what I want to know.

**Reply by zamotany** [reply]
Cool, keep in mind that by default the chunks generated that way are _remote chunks_ so you will need to use `ChunkManager` to load them or you can make them _local chunks_:
- for remote: https://re-pack.netlify.app/docs/api/react-native/classes/ChunkManager
- for local: https://re-pack.netlify.app/docs/api/node/interfaces/OutputPluginConfig#localchunks

**Comment by Ha0ran2001** [comment]
What about just use webpack's code splitting way https://webpack.js.org/guides/code-splitting/#entry-points
